launcher/minecraft/launch/DirectJavaLaunch.cpp

```diff
@@ -50,7 +55,32 @@ void DirectJavaLaunch::executeTask()
 
     auto javaPath = FS::ResolveExecutable(instance->settings()->get("JavaPath").toString());
 
-    m_process.setProcessEnvironment(instance->createEnvironment());
+    auto env = instance->createEnvironment();
+#ifdef Q_OS_LINUX
+    if (instance->settings()->get("EnableMangoHud").toBool())
+    {
+        auto preload = env.value("LD_PRELOAD", "") + ":libMangoHud_dlsym.so:libMangoHud.so";
+        auto lib_path = env.value("LD_LIBRARY_PATH", "") +  ":/usr/local/lib64/mangohud/:/usr/local/lib/mangohud/:/usr/lib64/mangohud/:/usr/lib/mangohud/";
+
+
+        env.insert("LD_PRELOAD", preload);
+        env.insert("LD_LIBRARY_PATH", lib_path);
+        env.insert("MANGOHUD", "1");
+
+        emit logLine("MANGOHUD", MessageLevel::Launcher);
+    }
+
+    if (instance->settings()->get("UseDiscreteGpu").toBool())
+    {
+        // Open Source Drivers    
+        env.insert("DRI_PRIME", "1");
+        // Proprietary Nvidia Drivers
+        env.insert("__NV_PRIME_RENDER_OFFLOAD", "1");
+        env.insert("__VK_LAYER_NV_optimus", "NVIDIA_only");
+        env.insert("__GLX_VENDOR_LIBRARY_NAME", "nvidia");
+    }
+#endif
+    m_process.setProcessEnvironment(env);
```

Comment:
Move this somewhere into `MinecraftInstance`.

It looks like `MinecraftInstance::createEnvironment()` is actually only intended for the pre/post-launch commands.

I would probably prefer if we just rename `MinecraftInstance::createEnvironment()` to `MinecraftInstance::createHookEnvironment()` and add a new method called `MinecraftInstance::createLaunchEnvironment()`

Maybe this:

```cpp
QProcessEnvironment MinecraftInstance::createLaunchEnvironment()
{
    // prepare the process environment
    QProcessEnvironment env = CleanEnviroment();

    // <insert performance options>

    return env;
}

```

Reply:
> Move this somewhere into `MinecraftInstance`.

Yeah that would be better

> It looks like `MinecraftInstance::createEnvironment()` is actually only intended for the pre/post-launch commands.

Is it? it looks more like a generic environment creation function for the startup process...

> I would probably prefer if we just rename `MinecraftInstance::createEnvironment()` to `MinecraftInstance::createHookEnvironment()` and add a new method called `MinecraftInstance::createLaunchEnvironment()`

I think instead of doing that we should leave the `createEnvironment` function alone and make a second one like this:

```cpp
QProcessEnvironment MinecraftInstance::createLaunchEnvironment()
{
    // prepare the process environment
    QProcessEnvironment env = createEnvironment();

    // <insert performance options>

    return env;
```

Reply:
The current `createEnvironment()` also adds env vars (`INST_NAME`, `INST_ID`, `INST_DIR`, ...) which shouldn't probably be passed to Minecraft. They probably don't make a functional difference, but along the lines of #606, we should avoid giving Microsoft more data points to uniquely identify users. On the other hand it might be handy for wrapper scripts. Not sure tbh

Reply:
I think its not within the scope of this PR to change that behavior, though this definetly should be looked at in the future.